shared/editor/plugins/PlaceholderPlugin.ts

```diff
@@ -25,24 +28,29 @@ export class PlaceholderPlugin extends Plugin {
 
   constructor(config: Config) {
     super({
+      state: {
+        init: (_, state: EditorState) => ({
+          decorations: this.createDecorations(state, config),
+        }),
+        apply: (tr, pluginState, oldState, newState) => {
+          // Only recompute if doc or selection changed
+          if (tr.docChanged || tr.selectionSet) {
+            return { decorations: this.createDecorations(newState, config) };
+          }
+          return pluginState;
+        },
+      },
       props: {
         decorations: (state) => {
-          const decorations: Decoration[] = map(
-            this.placeholders(state),
-            (placeholder) =>
-              Decoration.node(placeholder.from, placeholder.to, {
-                class: "placeholder",
-                "data-empty-text": placeholder.text,
-              })
-          );
-          return DecorationSet.create(state.doc, decorations);
+          const pluginState = this.getState(state);
+          return pluginState ? pluginState.decorations : null;
         },
       },
     });
     this.config = config;
```

Comment:
```suggestion
```

Reply:
nit: `private config: Config;` above the constructor can be removed. GitHub wouldn't let me suggest it the way I suggested previous change 😬